### PR TITLE
Move Game functionality to GameBoard component

### PR DESCRIPTION
**Pull request by beccarothwell**
Move all game functionality out of App and into GameBoard component to enable ease of isLoading conditional rendering.

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -1,313 +1,43 @@
...
-  const currentRoundRef = useRef<number>(0);
-  const currentPlayerRef = useRef<Player>(EMPTY_PLAYER);
+  const [game, setGame] = useState<Game>(PLACEHOLDER_GAME);
   const [isLoading, setIsLoading] = useState<boolean>(true);
 
   useEffect(() => {
     async function fetchPack() {
       const pack = await fetchPokemonPack();
       setIsLoading(false);
+      setIsLoading(false);
```
Comment (jennysivapalan) [inline client/src/App.tsx:23]: does this need to be in twice?

File: client/src/components/GameBoard/GameBoard.test.tsx
Comment (jennysivapalan) [inline client/src/components/GameBoard/GameBoard.test.tsx:1]: Are there tests coming or should we add this later?

Comment (jennysivapalan) [reply client/src/components/GameBoard/GameBoard.test.tsx:1]: We can do it later

**Review by jennysivapalan — APPROVED** [review]
Just one minor comment on the duplicate line

File: client/src/App.tsx (hunk 1 above)
Comment (beccarothwell) [reply client/src/App.tsx:23]: It does not, that's from a merge, will remove

**Pull request merged**